SystemInformer/delayhook.c

```diff
@@ -1451,33 +1714,359 @@ BOOL WINAPI PhSystemParametersInfoHook(
+    )
+{
+    static PH_INITONCE initOnce = PH_INITONCE_INIT;
+    static COLORREF colLinkNormal = RGB(0, 0 ,0);
+    static COLORREF colLinkHot = RGB(0, 0, 0);
+    static COLORREF colLinkPressed = RGB(0, 0, 0);
+    HWND WindowHandle;
+
+    if (PhBeginInitOnce(&initOnce))
+    {
+        HTHEME hTextTheme = PhOpenThemeData(NULL, VSCLASS_TEXTSTYLE, 0);
+        if (hTextTheme)
+        {
+            PhGetThemeColor(hTextTheme, TEXT_HYPERLINKTEXT, TS_HYPERLINK_NORMAL, TMT_TEXTCOLOR, &colLinkNormal);
+            PhGetThemeColor(hTextTheme, TEXT_HYPERLINKTEXT, TS_HYPERLINK_HOT, TMT_TEXTCOLOR, &colLinkHot);
+            PhGetThemeColor(hTextTheme, TEXT_HYPERLINKTEXT, TS_HYPERLINK_PRESSED, TMT_TEXTCOLOR, &colLinkPressed);
+            PhCloseThemeData(hTextTheme);
+        }
+        PhEndInitOnce(&initOnce);
+    }
+
+    if ((WindowHandle = WindowFromDC(hdc)) &&
+        (PhIsDarkModeAllowedForWindow(WindowHandle) || PhGetWindowContext(WindowHandle, TASKDIALOG_CONTEXT_TAG)))    // HACK
+    {
+        WCHAR windowClassName[MAX_PATH];
+        GETCLASSNAME_OR_NULL(WindowHandle, windowClassName);
+        if (PhEqualStringZ(windowClassName, WC_LINK, FALSE))
+        {
+            COLORREF color = GetTextColor(hdc);
+            if (color == colLinkNormal || color == colLinkHot || color == colLinkPressed)
+                SetTextColor(hdc, PhMakeColorBrighter(color, 95));
+        }
+    }
+    
+    return PhDefaultComCtl32DrawTextW(hdc, lpchText, cchText, lprc, format);
+}
+
+HRESULT PhGetThemeColorHook(
+    _In_  HTHEME   hTheme,
+    _In_  int      iPartId,
+    _In_  int      iStateId,
+    _In_  int      iPropId,
+    _Out_ COLORREF* pColor
+    )
+{
+    WCHAR className[MAX_PATH];
+
+    HRESULT retVal = PhDefaultGetThemeColor(hTheme, iPartId, iStateId, iPropId, pColor);
+
+    if (iPropId == TMT_TEXTCOLOR && iPartId == TDLG_MAININSTRUCTIONPANE)
+    {
+        if (PhGetThemeClass(hTheme, className, RTL_NUMBER_OF(className)) &&
+            PhEqualStringZ(className, VSCLASS_TASKDIALOGSTYLE, TRUE))
+        {
+            *pColor = PhMakeColorBrighter(*pColor, 150); // Main header.
+        }
+    }
+    else if (iPropId == TMT_TEXTCOLOR)
+    {
+        if (PhGetThemeClass(hTheme, className, RTL_NUMBER_OF(className)) &&
+            PhEqualStringZ(className, VSCLASS_TASKDIALOGSTYLE, TRUE))
+        {
+            *pColor = PhThemeWindowTextColor; // Text color for check boxes, expanded text, and expander button text.
+        }  
+    }
+
+    return retVal;
+}
+
+HTHEME PhOpenNcThemeDataHook(
+    _In_ HWND    hwnd,
+    _In_ LPCWSTR pszClassList
+    )
+{
+    if (PhEqualStringZ((PWSTR)pszClassList, VSCLASS_SCROLLBAR, TRUE) &&
+        PhIsDarkModeAllowedForWindow(hwnd))
+    {
+        return PhDefaultOpenNcThemeData(NULL, L"Explorer::ScrollBar");
+    }
+
+    return PhDefaultOpenNcThemeData(hwnd, pszClassList);
+}
+
+BOOLEAN CALLBACK PhInitializeTaskDialogTheme(
+    _In_ HWND WindowHandle,
+    _In_opt_ PVOID CallbackData
+)
```

Comment:
nit: noticing a few areas where the `)` isn't indented - don't worry about correcting them unless you end up having to make changes - @dmex or I can go in and clean them up after merging